### PR TITLE
[DataGrid] Fix pressing cursor down on an empty grid throws exception

### LINKED ISSUE #6446: [data grid] Pressing arrow down and page down on an empty grid throws exception.
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Selecting a column header and then pressing cursor down on an empty grid causes an exception. 

### Expected behavior 🤔

No exception should be thrown.

### Steps to reproduce 🕹

Steps:
1. Create a grid with columns but no row data.
2. Click a column header so it becomes outlined.
3. Press cursor down.
4. Exception is thrown:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'id')
    at useGridKeyboardNavigation.js:71:1
    at Object.current (useGridKeyboardNavigation.js:246:1)
    at EventManager.enhancedHandler (useGridApiEventHandler.js:39:1)
    at EventManager.emit (EventManager.js:71:1)
    at Object.publishEvent (useGridApiInitialization.js:36:1)
    at apiRef.current.<computed> [as publishEvent] (useGridApiMethod.js:14:1)
    at GridColumnHeaderItem.js:87:1
    at HTMLUnknownElement.callCallback (react-dom.development.js:4164:1)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4213:1)
    at invokeGuardedCallback (react-dom.development.js:4277:1)
```

### Context 🔦

Simple keyboard navigation.

### Your environment 🌎

_No response_

### Order ID 💳 (optional)

45466

### PR DESCRIPTION
**Pull request by Dekshut**
Fixes #6446

### CODE REVIEW

File: packages/grid/x-data-grid/src/hooks/features/keyboardNavigation/useGridKeyboardNavigation.ts [hunk 1]
```diff
@@ -89,7 +89,7 @@ export const useGridKeyboardNavigation = (
 
   const getRowIdFromIndex = React.useCallback(
     (rowIndex: number) => {
-      return currentPageRows[rowIndex].id;
+      return currentPageRows[rowIndex]?.id;
```
Comment (MBilalShafi) [inline packages/grid/x-data-grid/src/hooks/features/keyboardNavigation/useGridKeyboardNavigation.ts:92]:
```suggestion
      return currentPageRows[rowIndex].id;
```

File: packages/grid/x-data-grid/src/hooks/features/keyboardNavigation/useGridKeyboardNavigation.ts [hunk 2]
```diff
@@ -247,7 +247,7 @@ export const useGridKeyboardNavigation = (
 
       const viewportPageSize = apiRef.current.unstable_getViewportPageSize();
       const colIndexBefore = params.field ? apiRef.current.getColumnIndex(params.field) : 0;
-      const firstRowIndexInPage = 0;
+      const firstRowIndexInPage = currentPageRows.length > 0 ? 0 : null;
```
Comment (MBilalShafi) [inline packages/grid/x-data-grid/src/hooks/features/keyboardNavigation/useGridKeyboardNavigation.ts:250]:
```suggestion
      const firstRowIndexInPage = currentPageRows.length > 0 ? 0 : -1;
```

Comment (MBilalShafi) [reply packages/grid/x-data-grid/src/hooks/features/keyboardNavigation/useGridKeyboardNavigation.ts:250]: After doing this change you might update if conditions for cases `ArrowDown` and `PageDown` like so:

**ArrowDown:**
```diff
- if (firstRowIndexInPage !== null) {
+ if (firstRowIndexInPage >= 0) {
```

**PageDown:**
```diff
- if (firstRowIndexInPage !== null && lastRowIndexInPage !== null) {
+ if (firstRowIndexInPage >= 0 && lastRowIndexInPage >= 0) {
```

Comment (MBilalShafi) [reply packages/grid/x-data-grid/src/hooks/features/keyboardNavigation/useGridKeyboardNavigation.ts]: @Dekshut Please do above mentioned changes for `handleColumnHeaderKeyDown` to make it work properly

Comment (Dekshut) [reply packages/grid/x-data-grid/src/hooks/features/keyboardNavigation/useGridKeyboardNavigation.ts]: I guess I did. Please check it again

**Pull request closed without merging**